File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character, Character> hashmap = new HashMap<>();
        
        
        for(int i = 0; i < s.length(); i++) {
            char a = s.charAt(i);
            char b = t.charAt(i);
            
            if(hashmap.containsKey(a)) {
                if(hashmap.get(a).equals(b)) {
                    continue;
                }
                else {
                    return false;
                }
            }
            else {
                if(!hashmap.containsValue(b)) {
                    hashmap.put(a, b);
                }
                else {
                    return false;
                }
            }
        }
        return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        map1 = {}
        map2 = {}
        s1 = 0
        t1 = 0

        for st in s:
            if st not in map1:
                if t[t1] in map2 and map2[t[t1]] != st:
                    return False
                map1[st] = t[t1]
                
                map2[t[t1]] = st
            else:
                if map1[st] != t[t1]:
                    return False
            t1 += 1
        return True
